src/pint/utils.py

```diff
@@ -434,3 +435,115 @@ def show_param_cov_matrix(matrix, params, name="Covaraince Matrix", switchRD=Fal
     contents = output.getvalue()
     output.close()
     return contents
+
+
+def dmxparse(fitter, save=False):
+    """Run dmxparse in python using PINT objects and results.
+
+    Based off dmxparse by P. Demorest (https://github.com/nanograv/tempo/tree/master/util/dmxparse)
+
+    Inputs:
+    :param fitter: PINT fitter used to get timing residuals, must have already run GLS fit
+    :param save: boolean; if True saves output to text file in the format of the TEMPO version.
+    If not output save file is desired, save = False (which is the default)
+    Output file name is dmxparse.out
+    """
+    # We get the DMX values, errors, and mjds (same as in getting the DMX values for DMX v. time)
+    # Get number of DMX epochs
+    dmx_epochs = 0
+    for p in fitter.model.params:
+        if "DMX_" in p:
+            dmx_epochs += 1
+    # Get DMX values (will be in units of 10^-3 pc cm^-3)
+    DMX_keys = []
+    DMXs = []
+    DMX_Errs = []
+    DMX_R1 = []
+    DMX_R2 = []
+    DMX_center_MJD = []
+    for ii in range(1, dmx_epochs + 1):
+        if ii < 10:
+            ext = "000%s" % (ii)
+        elif ii >= 10 and ii < 100:
+            ext = "00%s" % (ii)
+        elif ii >= 100 and ii < 1000:
+            ext = "0%s" % (ii)
+        DMX_keys.append("DMX_%s" % (ext))
```

Comment:
This can be written more concisely like this:
```
    DMX_keys.append("DMX_{:04d}".format(ii))
```
and similarly for the next few lines that use `ext`

Reply:
Nice.  I was going to make the same comment, but you beat me to it.